### PR TITLE
add create widget option to profile

### PR DESCRIPTION
**Pull request by sarveshpro**
Changes in this pull request:
- add create widget option to profile

Environment variable required - `WIDGET_URL`

@norbertschuler please review private to public account changing modal text.

### CODE REVIEW

File: src/features/user/UserProfile/components/EmbedModal.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            isPrivate: false
+        }
+        if (!isLoading && token) {
+            try {
+                putAuthenticatedRequest(`/app/profile`, bodyToSend, token).then((res) => {
+                    setSeverity('success')
+                    setSnackbarMessage(ready ? t('editProfile:profileSaved') : '')
+                    setEmbedModalOpen(false)
+                    setIsUploadingData(false)
+                    router.push(`${process.env.WIDGET_URL}?user=${userprofile.id}&tenantkey=${process.env.TENANTID}`);
```
Comment (norbertschuler) [inline src/features/user/UserProfile/components/EmbedModal.tsx:82]: There is a bug in our TreeMap and also in our TreeProfile widget that it is crashing if a `TENANTID` is given which does not exist in https://github.com/Plant-for-the-Planet-org/planet-widgets/blob/develop/tenant.json
e.g. https://widgets.plant-for-the-planet.org/?user=prf_KL79s31SQaotgohmtXiETjcq&tenantkey=ten_NxJq55pm

As we cannot always add all possible tenants in the widgets and wrong `tenantkey` parameters can be specified, the widget code must be failsafe for non existing `tenantkey` parameters given!

Comment (norbertschuler) [reply src/features/user/UserProfile/components/EmbedModal.tsx:82]: I added a fix for the widgets with https://github.com/Plant-for-the-Planet-org/planet-widgets/pull/49

**Review by norbertschuler — APPROVED** [review]
Works well for me, but https://github.com/Plant-for-the-Planet-org/planet-widgets/pull/49 needs to be fixed before this got released.

File: src/features/user/UserProfile/components/EmbedModal.tsx (hunk 1 above)
Comment (harshvitra) [reply src/features/user/UserProfile/components/EmbedModal.tsx:82]: I merged the PR on the widgets repo

Comment (norbertschuler) [reply src/features/user/UserProfile/components/EmbedModal.tsx:82]: Also must be released before we release this :-)

**Pull request merged**